src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -134,6 +134,12 @@ public class Popup : Control, IPopupHostProvider
         public static readonly StyledProperty<bool> TopmostProperty =
             AvaloniaProperty.Register<Popup, bool>(nameof(Topmost));
 
+        /// <summary>
+        /// Defines the <see cref="TakesFocusFromNativeControl"/> property.
+        /// </summary>
+        public static readonly AttachedProperty<bool> TakesFocusFromNativeControlProperty =
```

Comment:
Do we need `TakesFocusFromNativeControl` to be an attached property?

It's only used and respected by `ContextMenu`, which already calls `AddOwner` on several `Popup` properties. For consistency, I believe we should do the same here.

Reply:
I did mention this in the description, but it's easy to miss:

> it's attached because you can set it directly on Popup or on a control that uses a popup - only ContextMenu respects this property currently

Reply:
It makes sense. Thanks!